libs/books/data-access/src/lib/+state/reading-list.effects.ts

```diff
@@ -54,9 +66,74 @@ export class ReadingListEffects implements OnInitEffects {
     )
   );
 
+  undoAddtoReadingList$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(ReadingListActions.confirmedAddToReadingList),
+      filter(({book}) => book.isOpenSnackBar),
+      switchMap(({ book }) => {
+        return of(
+          ReadingListActions.openSnackBar({
+            item: { ...book, bookId: book.id, isOpenSnackBar: false },
+            message: `${book.title}: added to reading list`,
+            isAdded: true
+          })
+        );
+      })
+    )
+  );
+
+  undoRemoveFromReadingList$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(ReadingListActions.confirmedRemoveFromReadingList),
+      filter(({item}) => item.isOpenSnackBar),
+      switchMap(({ item }) => {
+        return of(
+          ReadingListActions.openSnackBar({
+            item: { id: item.bookId, ...item, isOpenSnackBar: false },
+            message: `${item.title}: removed from reading list`,
+            isAdded: false
+          })
+        );
+      })
+    )
+  );
+
+  @Effect({ dispatch: false })
+  openSnackBar$ = this.actions$.pipe(
```

Comment:
Please remove this code from here as it will not required once you implement above comments.